src/vanilla.ts

```diff
@@ -244,6 +244,7 @@ const buildProxyFunction = (
     const handler: ProxyHandler<T> = {
       deleteProperty(target: T, prop: string | symbol) {
         const prevValue = Reflect.get(target, prop)
+        snapCache.delete(target as Object)
```

Comment:
We never delete the cache elsewhere, so it doesn't feel very nice.
Do you happen to know why deleteProperty doesn't increment the version number?
I think we should increment it if it doesn't.

Reply:
Ok, I will try more on that way.
And I found out that although if I increase the target version of deleted property in `deleteProperty` it is still getting the same version in [here](https://github.com/pmndrs/valtio/blob/d1da1b234ce8d4b0460a14d83e16a8cfa5a94bcd/src/vanilla.ts#L105-L114).

Reply:
> Would you be able to add a failing test?
Do you mean that adding a test fails when [this symptom]((https://github.com/pmndrs/valtio/issues/712) has occurred?

Reply:
ah, yeah. adding a test to represent #712 issue, which fails in the current main branch, and will be resolved with this PR.

Reply:
> it is still getting the same version

Is it like reading a stale version from somewhere?